### PR TITLE
cirrus lib.sh: refactor req_env_var()

### PR DESCRIPTION
**Pull request by edsantiago**
Existing code was not working due to a bash gotcha ('exit'
from a pipeline). It also had unnecessary duplication.

New version is safer; also includes unit tests.

Existing invocations of req_env_var replaced via:

   $ [ edit setup_environment.sh, move one closing quote to its own line ]
   $ perl -ni -e 's/(?<=req_env_var )"(\S+)\s+\$\1"/$1/; if (/req_env_var "$/ .. /^\s*"/) { chomp; s/(?<=\S)\s.*//; if (/^\s*"/) { print "\n" } else { unless (/req_env_var/) { s/^\s+//; print " ";} print;} } else { print }' $(ack -l req_env_var)
   $ [ hand-massage an incorrect instance of '@' in lib.sh:ircmsg() ]

Signed-off-by: Ed Santiago <santiago@redhat.com>

### CODE REVIEW

**Comment by edsantiago** [comment]
Sorry; I have no hints for reviewing other than suggesting the use of a good diff colorizer.

File: contrib/cirrus/lib.sh [hunk 1]
```diff
@@ -97,20 +109,14 @@ PACKER_BUILDS $PACKER_BUILDS
...
+    req_env_var UNSET_ENV_VARS
     echo "Unsetting $(echo $UNSET_ENV_VARS | wc -w) environment variables"
     unset -v UNSET_ENV_VARS $UNSET_ENV_VARS || true  # don't fail on read-only
 }
 
 die() {
-    req_env_var "
-        1 $1
-        2 $2
-    "
```
Comment (cevich) [inline contrib/cirrus/lib.sh]: Oh good you caught this possible inf. loop :smile:

File: contrib/cirrus/lib.sh.t [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+# ...but error if any single desired one is unset
+test_rev "FOO BAR" 9 'FATAL: test_rev() requires $BAR to be non-empty'
+
+# ...and OK if all args are set
+BAR=1
+test_rev "FOO BAR" 0 ''
+
+###############################################################################
+
+exit $rc
```
Comment (cevich) [inline contrib/cirrus/lib.sh.t]: ooooohhhhhhh praise $DIETY thank theee for blessing @edsantiago to write this thy unittest thanks be the $CI

**Comment by cevich** [comment]
LGTM pending tests pass with something like this:

```diff
diff --git a/Makefile b/Makefile
index 9228ec71..cf4e9364 100644
--- a/Makefile
+++ b/Makefile
@@ -188,6 +188,7 @@ localunit: test/goecho/goecho varlink_generate
                --tags "$(BUILDTAGS)" \
                --succinct
        $(MAKE) -C contrib/cirrus/packer test
+       bash contrib/cirrus/lib.sh.t
 
 ginkgo:
        ginkgo -v -tags "$(BUILDTAGS)" $(GINKGOTIMEOUT) -cover -flakeAttempts 3 -progress -trace -noColor -nodes 3 test/e2e/.
```

Other option is to add it to `.cirrus.yml` as part of the gate task.  However it's done, it would be great to have it continuously validated.  Thanks again for adding those tests, that really means a lot to me.

**Comment by edsantiago** [comment]
Force-pushed with requested Makefile change. Thanks for the suggestion.

**Comment by cevich** [comment]
@edsantiago fantastic, thanks.

LGTM

@mheon may we fast-track this one please, it adds some sorely needed fixes *AND* actual CI-unittests for a heavily used function in our Cirrus scripts.  It's also needed for my #2561

**Comment by cevich** [comment]
/me sprinkles some go-faster and run-clean dust onto the automation pixies

**Comment by mheon** [comment]
My bash-fu is very weak, but this LGTM

**Comment by cevich** [comment]
Thanks Matt and Tom.

/lgtm

**Comment by edsantiago** [comment]
@mheon (and future maintainers) the key background for this is that #2561 is making some changes to various invocations of the `req_env_var` function. While reviewing I noticed that the function was not working as intended; but fixes to it would be too cumbersome to review as part of that PR. If 2561 is rebased on top of this, it should (I hope) be easier to review.

**Pull request merged**